### ISSUE TITLE
DZGUI error code 100 in desktop mode. 

### ISSUE THREAD

**Issue by BennyH0nda** [aclist/dztui#31]
**Info**
DZGUI version: 2.6.0
Mode (normal/debug): stable
Server ID:
Linux distribution:

**Bug description**

I have been using DzGUI to play on my modded server and it’s been working great until last night.  I keep getting this error message when I try to launch it.  It doesn’t even launch in gaming mode. I’ve tried to reinstall it and erase the configure files with no luck.  I’ve change the compatibility mode to photon experimental as well.  When I use the console and reinstall it, the launcher works and I can connect to the server but I am not able to launch it normally. 

**To Reproduce**
1. Go to steam library in the desktop mode (gaming mode doesn’t show the error. It just crashes)
2. Click on DZGUI icon
3. See error

**Expected behavior**
The GUI should load with options to pick a server and start playing on a modded server. 
![8049E2C0-B32C-4F38-A51C-A7E85AAE5896](https://user-images.githubusercontent.com/114331610/192133526-17cc2118-454c-4926-9784-5d5b1c732384.jpeg)

**Comment by aclist** [comment]
There have been no recent changes to the stable version of DZGUI. 

There is a bug in KDE Plasma in the way it handles shortcuts that point to a shell script. This error should not occur if you launch the correct shortcut. 

1. When you added it to Steam, did you add the `.desktop` shortcut as a non-steam game, or the `dzgui.sh` itself?
2. If you are trying to do a clean reinstall, you need to remove everything in `$HOME/.local/share/applications/` and `$HOME/.local/share/dzgui` as well.

Did you make any other changes before this error started to show up?

**Comment by BennyH0nda** [comment]
Thank you for responding so quickly!   I added the shortcut as a non-steam game but I’ll double check out removing everything when I do another fresh install.   The only changes I made after getting it to work the first time was changing the icons in the steam library.   It’s strange because I can get it to run when I use the install command in the konsole that one time but it doesn’t work after that.

**Comment by aclist** [comment]
I try to check bug reports at least 1-2 times daily.

I haven't been able to reproduce your problem, but will try reinstalling again on Steam Deck and keep analyzing. 

Has this problem always been present, or did it start appearing after upgrading versions?

**Comment by aclist** [comment]
Sorry, accidentally pushed the "close issue" hotkey. Bug remains open, feel free to submit additional updates.

**Comment by BennyH0nda** [comment]
It hasn’t always been a problem. I’ve gotten it to work for a few days until last night.  I don’t remember doing anything besides changing the icons in the steam library.   It might be something to do with my shortcut because I can get it to work when I reinstall it in the terminal.  The very last thing I did was adding a favorite server and then it stopped working for me.

**Comment by aclist** [comment]
Hmm, none of those things should affect the shortcut per se, unless there was some undocumented change to the OS. I'm thinking about it. 

To reinstall, I'd recommend following the "Automatic method" listed here: https://aclist.github.io/dzgui/dzgui.html#_preparation

If you invoke that once from the terminal, the rest of the setup should be automatic and it will create new versions/shortcuts for everything in the desired locations.

**Comment by aclist** [comment]
I updated my comment above. If you perform a reinstall, you need to remove `$HOME/.local/share/applications/dzgui.desktop` and `$HOME/.local/share/dzgui`. But you shouldn't have to manually touch those, as they get overwritten whenever you install or update the application.

If you don't want to reinstall, you can just change the version number at the top of `$HOME/.dzgui/dzgui.sh` to another number: e.g., `2.5.0` and it will trigger an update when you launch it again (which replaces all of the necessary files).

You should also right-click the desktop icon, select Properties in the dialog, then the Application pane. The `command` field should point to `/home/deck/.local/share/dzgui/dzgui.sh`. 

We could look at including an additional shortcut or start menu option with the application that could be used to "force reinstall," if that seems convenient.

**Comment by aclist** [comment]
OK, I was able to reproduce and fix the problem. Like I said, this comes back to an issue with the Freedesktop spec and .desktop files. They have to be formatted in a very specific way. In this case, if there is an extra line at the top of the script, it fails, as it expect a line calling the interpreter there.

I updated the script with a hotfix and bumped the version to `2.6.1`. If you install or update now, the desktop shortcut should be fixed.
